.github/CODEOWNERS

```diff
@@ -674,7 +674,6 @@
 # TODO(#11811): Replace @BenHenning with @seanlip after 2021-02-07 for the next
 # 8 lines.
 /core/templates/services/svg-sanitizer.service.ts @BenHenning
-/scripts/linters/warranted_angular_security_bypasses.py @BenHenning
```

Comment:
Hmm in retrospect this seems problematic. Isn't it the case that anyone can work around the bypass just by adding a suppression lint check? That doesn't seem right. The current setup forces the author to change the bypasses file since otherwise they will fail CI, and we can use codeowners to make sure the exception is valid. Allowing any line of code opt out of the bypass change prevents us from locking this down, which isn't ideal given it can introduce security issues.

Is there a way to centralize the whitelist with the new approach?

Reply:
@BenHenning I wanted to add some notes here:

Even the existing implementation allows developers to bypass using `disable-bad-pattern-check` flag or adding file/dir name in the general_purpose_linter.py file

Currently, we have some rules (BR) to disallow usage of the disable//bypass flag for some specific rules [R] which disallows developers to bypass some rules. But those "BR" checks can also be bypassed:

// eslint-disable [R] (BR)

This will disable both the R and BR rules^

In May, the linter team is going to implement a feature through which a bot will add comments in the PR thread if someone uses a disable/bypass flag and will ask them to provide a reason for the usage, which reviewers can go through!

Eslint doesn't allow a way to disallow usage of bypass. Maybe have a strict regex check in python which will disallow bypassing some specific rules (for current use and future use)?  


Let me know if you have any other idea/suggestion?

Reply:
Thanks for the additional context @DubeySandeep. A regex check might be a better fit. I think the important thing here is that we can ensure exceptions to security-sensitive checks require codeowner approval in a way that can't be avoided. Even though the previous approach still supported a workaround, one advantage is that it defaulted to an allow-list based approach. It seems like this approach removes that outright which means it falls on all codeowners to ensure the bypass is allowed when it's allowed. Where possible, we should try to retain the existing approval structure for these types of exceptions.

What do you think we should do here?

Reply:
[I'm working on this with the linter team to find a general solution, will update this thread once we have an approach to fix it]

Reply:
I look into possible ways to disallow it with eslint but looks like it won't be possible to do so. @AdityaDubey0 Can you please write a simple python script to disallow the bypass comment (And should only exist in files mentioned in warranted_angular_security_bypasses.py file).